app/src/test/webapp/app/error/error.ctrl.test.js

```diff
@@ -178,4 +193,68 @@ describe("ErrorCtrl", function() {
         });
     });
   });
+
+  describe("success after calling resubmit", function() {
+    it("should display a success message when the message is resubmitted", function() {
+      spyOn(messageCenterService, 'add');
+
+      $controller("ErrorCtrl", {
+        $scope: $scope
+      });
+
+      $scope.message = resubmittableMessage;
+
+      $scope.resubmitMessage();
+      $scope.$apply();
+
+      expect(messageCenterService.add).toHaveBeenCalledWith('success', 'Resubmit successful!', { "timeout": 3000 });
+    });
+  });
+
+  describe("failure after calling resubmit", function() {
+    beforeEach(function resubmitWithServerProblem() {
+      msgSvc.resubmitMessage = function() {
+        return $q.when({
+          data: {
+            "status": "Failure",
+            "errorMessage": "Just failed"
+          }
+        });
+
+      };
+    });
+
+    it("should display a failure message when a message isn't resubmittable", function(){
+
+      spyOn(messageCenterService, 'add');
+
+      $controller("ErrorCtrl", {
+        $scope: $scope
+      });
+
+      $scope.message = testMessage;
```

Comment:
Also, what makes a message resubmittable or not? Would be nice to actually just see the data here inline I think.